### ISSUE TITLE
Support plural resource nouns

### ISSUE THREAD

**Issue by mattmoor** [knative/client#812]
### Feature request

In `kubectl` and `tkn` I can use either the singular or plural for resource names, but with `kn` I get an error when using the plural:

```shell
kn services list
Error: unknown command 'services'
Run 'kn --help' for usage.
```

IMO `services` should simply alias `service`.

**Comment by rhuss** [comment]
+1 especially for commands that operate on multiple resource instances (but I wouldn't also mind so support both variants unconditionally eg. also `kn services create`)

**Comment by navidshaikh** [comment]
should we also consider bringing other aliases with this? 

for example
 - `svc` for services
 - `rev` for revisions
etc

**Comment by duglin** [comment]
svc might be too easily confused with k8s svc

**Comment by rhuss** [comment]
> svc might be too easily confused with k8s svc

don't think this is a strong argument as the same applies for 'service' itself, too. (i.e. that it can be confused with a k8s 'service')

I thin we can consider command shortcuts but I would do it one step after the other. @navidshaikh would you mind to open an extra issue for this ? There we could also collect the possible shorcuts (like also `src` for `source` ...)

**Comment by mattmoor** [comment]
To chime in on shortnames, muscle memory would have me typing `ksvc`, which is how we alias it in kubectl 😇

**Comment by rhuss** [comment]
Fixed with 0.18
